tests/test_operation_builder.py

```diff
@@ -499,8 +499,12 @@ class SuccessorOp(IRDLOperation):
 
 
 def test_successor_op_successor():
-    block = Block()
-    op = SuccessorOp.build(successors=[block])
+    block0 = Block()
+    op = SuccessorOp.build(successors=[block0])
+
+    block1 = Block([op])
+    _ = Region([block1])
```

Comment:
Ah that it belongs to a region...

Reply:
I'd recommend adding a comment, or doing something like this to make it a bit more like real-world code:
``` python
with ImplicitBuilder(Block())):
  op = SuccessorOp.build(successors=[block0])
```

Reply:
I'll add a comment but I feel improving these tests (i.e., using the `ImplicitBuilder`) is outside the scope of this PR.
This can be something separate misc PR later.

Reply:
Comment is already super helpful, thank you

Reply:
> Ah that it belongs to a region...

My understanding is that this tests checks that you can build an operation with successors with IRDL.
The fact that it needs to be in a block and a region is an effect of this PR but it is tested in `test_ir.py` and `test_builtin_traits.py`.